### PR TITLE
more re-exports and use EnumOrLiteral from core

### PR DESCRIPTION
**Pull request by yen-tt**
export Address and ComplexUrl interface from core as well
remove EnumOrLiteral from headless, and use the one from core instead. 

J=SLAP-2324

### CODE REVIEW

**Review by oshi97 — APPROVED** [review]

File: src/search-core-re-exports.ts [hunk 1]
```diff
@@ -44,10 +44,13 @@ export {
   IosAppUrlDirectAnswer,
   AndroidAppUrlDirectAnswer,
   ComplexUrlDirectAnswer,
+  ComplexURL,
```
Comment (nmanu1) [inline src/search-core-re-exports.ts:47]: can these exports be re-organized a little? I think we've been trying to keep them mostly alphabetical, but maybe the DA stuff can all go together at the end or something

Comment (yen-tt) [reply src/search-core-re-exports.ts]: sure, updated.

File: src/search-core-re-exports.ts [hunk 2]
```diff
@@ -91,5 +68,34 @@ export {
...
+  DirectAnswer,
+  DirectAnswerType,
+  BuiltInFieldType,
+
+  BaseFeaturedSnippetDirectAnswer,
+  FeaturedSnippetDirectAnswer,
+  MultiLineTextSnippetDirectAnswer,
+  RichTextSnippetDirectAnswer,
+
+  BaseFieldValueDirectAnswer,
```
Comment (nmanu1) [inline src/search-core-re-exports.ts:82]: nit: can these be alphabetized?

Comment (yen-tt) [reply src/search-core-re-exports.ts:85]: sure

File: src/search-core-re-exports.ts [hunk 3]
```diff
@@ -91,5 +68,34 @@ export {
...
+  FieldValueDirectAnswer,
+  InstagramHandleDirectAnswer,
+  IntegerDirectAnswer,
+  IosAppUrlDirectAnswer,
+  PhoneDirectAnswer,
+  RichTextDirectAnswer,
+  TextDirectAnswer,
+  TwitterHandleDirectAnswer,
+  UnknownFieldValueDirectAnswer,
+  UrlDirectAnswer,
```
Comment (nmanu1) [inline src/search-core-re-exports.ts:100]: nit: extra comma

**Review by nmanu1 — APPROVED** [review]

**Pull request merged**